File: FrontEnd/src/app/model/enrollment.ts
import { Company } from './company';

export class Enrollment {
  id: number = 0;
  company?: Company;
  enrollmentDate?: Date;

  constructor(id: number, company: Company, enrollmentDate: Date) {
    this.id = id;
    this.company = company;
    this.enrollmentDate = enrollmentDate;
  }
}

File: BackEnd/src/main/java/com/jee/faang/models/metier/Enrollment.java
package com.jee.faang.models.metier;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDate;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Enrollment {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @OneToOne
    private Guest guest;
    @OneToOne
    private Company company;
    private LocalDate enrollmentDate;
}
